### PR TITLE
Ultra-compact mode ui bug fixes #2576

### LINKED ISSUE #2576: Ultra compact bugs🐛
Some minor visual bugs in the ultra compact mode. If more shows up before these have been fixed, they will be added to this list.

Buttons misaligned while logging in:
Image

Breadcrumbs:
Image

Warning in the console when opening the front page (and a few other similar warnings about isCompact on other pages):
Image

Warning in the console when clicking "Pick template" in the query view:
Image

### PR DESCRIPTION
**Pull request by matusmlichsk**
[//]: # (Request Template Witsml Explorer)

[//]: # (Thank you for contributing to WITSML Explorer! Before submitting this PR, please fill in the following template.)

## Fixes

[//]: # (Write the GitHub issue number starting with #, or, for Equinor only, the Jira issue number starting with WE-)

This pull request fixes issue #2576 

## Description

[//]: # (Please include a written summary of the changes and which issue is fixed. List any dependencies that are required for this change._)

- tabs in query editor are fixed height without overflow
- data grid switcher position changed
- no console warns connected with transient styled component props
- isLoading button miss-align on login fixed
- breadcrumbs properly aligned

## Type of change

[//]: # (Mark any of the types of change that apply.)

* [x] Bugfix
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Enhancement of existing functionality
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## Impacted Areas in Application

[//]: # (List general components of the application that this PR will affect)

* [x] Frontend
* [ ] API
* [ ] WITSML
* [ ] Desktop
* [ ] Other (please describe)

## Checklist:

[//]: # (Please tick all the boxes or remove the ones that aren't needed and explain why)

*Communication*
* [ ] I have made corresponding changes to the documentation
* [ ] PR affects application security

*Code quality*
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [ ] New code is covered by passing tests

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: Src/WitsmlExplorer.Frontend/components/ContentViews/QueryView/components/QueryOptions/DataGridSwitch/DataGridSwitch.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  const isCompact =
+    useOperationState().operationState.theme === UserTheme.Compact;
+
+  return (
+    <Stack direction="row" alignItems="center" justifyContent="flex-start">
+      <Switch
+        checked={dataGridActive}
+        onChange={onClick}
+        size={isCompact ? "small" : "default"}
+        label="Use data grid view"
```
Comment (eliasbruvik) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/QueryView/components/QueryOptions/DataGridSwitch/DataGridSwitch.tsx:27]: Hard to read in dark mode: 
image

File: Src/WitsmlExplorer.Frontend/components/Sidebar/SearchFilter/SearchFilter.tsx [hunk 2]
```diff
@@ -187,13 +187,13 @@ const SearchFilter = (): ReactElement => {
   );
 };
 
-const SearchField = styled(TextField)<{ colors: Colors; isCompact: boolean }>`
+const SearchField = styled(TextField)<{ colors: Colors; $isCompact: boolean }>`
```
Comment (eliasbruvik) [inline Src/WitsmlExplorer.Frontend/components/Sidebar/SearchFilter/SearchFilter.tsx:190]: I'm getting a syntax warning for an extra } at the end of SearchField.

File: Src/WitsmlExplorer.Frontend/components/Breadcrumbs.tsx
Comment (eliasbruvik) [inline Src/WitsmlExplorer.Frontend/components/Breadcrumbs.tsx:1]: Seems like the problem with the breadcrumbs happens on both my monitors if I select 100% scaling, but it looks good on 125%. Adding this to the style in EdsBreadcrumbs.Breadcrumb makes it a bit better, but you might have a better idea.
```
lineHeight: "16px",
display: "inline-flex",
```

How it looks on 100%:
image

**Review by eliasbruvik — CHANGES_REQUESTED** [review]

File: Src/WitsmlExplorer.Frontend/components/Sidebar/SearchFilter/SearchFilter.tsx (hunk 2 above)
Comment (matusmlichsk) [reply Src/WitsmlExplorer.Frontend/components/Sidebar/SearchFilter/SearchFilter.tsx:190]: fixed

File: Src/WitsmlExplorer.Frontend/components/ContentViews/QueryView/components/QueryOptions/DataGridSwitch/DataGridSwitch.tsx (hunk 1 above)
Comment (matusmlichsk) [reply Src/WitsmlExplorer.Frontend/components/ContentViews/QueryView/components/QueryOptions/DataGridSwitch/DataGridSwitch.tsx:27]: fixed

File: Src/WitsmlExplorer.Frontend/components/Breadcrumbs.tsx
Comment (matusmlichsk) [reply Src/WitsmlExplorer.Frontend/components/Breadcrumbs.tsx:1]: should be fixed

**Review by eliasbruvik — APPROVED** [review]
*Reviewer checklist*
- [x] Read issue and PR
- [x] Pulled branch and manually tested
- [x] Verify that PR resolves issue
- [x] Reviewed the code

**Pull request merged**